src/ecs/components/audio/audio_emitter.rs

```diff
@@ -0,0 +1,31 @@
+use std::io::Cursor;
+
+use rodio::{Decoder, SpatialSink};
+
+use audio::Source;
+use audio::output::Output;
+use ecs::{Component, VecStorage};
+
+/// An audio source, add this component to anything that emits sound.
+pub struct AudioEmitter {
+    pub(crate) sink: SpatialSink,
+}
+
+impl AudioEmitter {
+    /// Creates a new AudioEmitter component initialized to the given positions.  These positions will stay synced with Transform
+    /// if the Transform component is available on this entity.
+    pub fn new(output: &Output, emitter_position: [f32; 3], left_ear: [f32; 3], right_ear: [f32; 3]) -> AudioEmitter {
+        AudioEmitter {
+            sink: SpatialSink::new(&output.endpoint, emitter_position, left_ear, right_ear),
+        }
+    }
+
+    /// Adds an audio source to the queue of sounds to be played from this emitter.
+    pub fn append(&self, source: &Source) {
```

Comment:
Current architecture doesn't permit multiple sounds at once, but it probably should.  I'll think about ways to accomplish this.